tests/unit/tuner/callback/test_model_checkpoint.py

```diff
@@ -0,0 +1,102 @@
+import os
+
+import numpy as np
+import pytest
+import torch
+import tensorflow as tf
+
+from finetuner.tuner.base import BaseTuner
+from finetuner.tuner.callback import BestModelCheckpoint, TrainingCheckpoint
+from finetuner.tuner.pytorch import PytorchTuner
+from finetuner.tuner.keras import KerasTuner
+from finetuner.tuner.state import TunerState
+
+
+@pytest.fixture(scope='module')
+def pytorch_model() -> BaseTuner:
+    embed_model = torch.nn.Linear(in_features=10, out_features=10)
+
+    return embed_model
+
+
+@pytest.fixture(scope='module')
+def keras_model() -> BaseTuner:
+    embed_model = tf.keras.Sequential(
+        [
+            tf.keras.layers.Flatten(input_shape=(28, 28)),
+            tf.keras.layers.Dense(10, activation='relu'),
+        ]
+    )
+    return embed_model
```

Comment:
Use only pytorch here, since the functionality you are testing does not depend on the framework

Reply:
I specifically used both because keras saves in a directory and pytorch saves in a file. So I use two ways to delete os.remove and shutil.rmtree I created two tests for pytorch and keras to be able to test both deletions

Reply:
Got it